### DISCUSSION TITLE
Node http api response getting late due to use socket.io module

### DISCUSSION THREAD

**Discussion by kapillms** [socketio/socket.io#4405] [Q&A]
When we include socket.io module in nodejs app the response time of rest node api is getting late (4 seconds), even no long computation or blocking code used in api

**Comment by darrachequesne** [comment]
Hi! As usual, are you able to reproduce the issue? Could you please provide some code?

**Reply by kapillms** [reply]
------------------------index.js--------------------------
const express = require('express'),
    https = require('https'),
    http = require('http'),
    fs = require('fs'),
    app = express(),
    path = require('path'),
    bodyParser = require('body-parser'),
    errorhandler = require('errorhandler'),
    routes = require('./services/routes'),
    routesv1 = require('./services/v1/routes'),
    jobs = require('./services/jobs/jobs'),
    var logger = require('./logger.js');
const options = {
    key: fs.readFileSync('./t.key'),
    cert: fs.readFileSync('./t.crt')
};
app.set('view', __dirname + '/views');
app.set('view engine', 'jade');
app.use(bodyParser.urlencoded({
    extended: true
}));
app.use(bodyParser.json({
    limit: '100kb'
}));

app.use(errorhandler());
app.use('/', express.static(path.join(__dirname, '/')));



app.get('/socket', function (req, res) {
    res.sendFile(__dirname + '/services/index.html');
});
const server = https.createServer(app,options).listen(3001);
server.setTimeout(900000);
var io = require('socket.io').listen(server);
global['socket'] = io;
routes(app,io);
routesv1(app,io);


process.on('uncaughtException', function (err) {
 //   logger.error(err);
});
const socket = require('./services/messaging');
socket(io);


------------------------------------------messaging.js-----------------------------------------------
module.exports = function (io) {
    
    io.sockets.on('connection', function (socket) {
        socket.on('startTyping', function(data,callBack) {
            try{
                io.sockets.to(data.user).emit('startTyping', data )
                return callBack({"success": 1,"msg": '' });

            }catch(error){
                return callBack({"success": 0,"msg":error.message });
            }
        })
       
    });
}
